qdrant_client/local/distances.py

```diff
@@ -75,6 +89,41 @@ def calculate_distance(
         raise ValueError(f"Unknown distance type {distance_type}")
 
 
+def calculate_best_scores(
+    query: RecoQuery, vectors: np.ndarray, distance_type: models.Distance
+) -> types.NumpyArray:
+    
+    def get_best_scores(examples: List[types.NumpyArray]) -> types.NumpyArray:
+        vector_count = vectors.shape[0]
+        
+        # get all positive scores
+        scores: List[types.NumpyArray] = []
+        for example in examples:
+            score = calculate_distance(example, vectors, distance_type)
+            scores.append(score)
+            
+        # keep only max (or min) for each vector
+        if distance_to_order(distance_type) == DistanceOrder.BIGGER_IS_BETTER:
+            if len(scores) == 0:
+                scores.append(np.full(vector_count, -np.inf))
+            best_scores = np.array(scores, dtype=np.float32).max(axis=0)
+        else:
+            if len(scores) == 0:
+                scores.append(np.full(vector_count, np.inf))
+            best_scores = np.array(scores, dtype=np.float32).min(axis=0)
+
+        return best_scores
+    
+    pos = get_best_scores(query.positive)
+    neg = get_best_scores(query.negative)
+    
+    # choose from best positive or best negative
+    if distance_to_order(distance_type) == DistanceOrder.BIGGER_IS_BETTER:
+        return np.where(pos > neg, pos, -(neg*neg))  
+    else:
+        return np.where(pos < neg, pos, neg*neg)
```

Comment:
these np.where statements require comments (at least for neg case)

Reply:
Added comments